cookbooks/cdo-varnish/templates/default/default.vcl.erb

```diff
@@ -1,231 +1,30 @@
-    int is_supported = 0;
-
-    /* We want to match 'zh-cn' and 'zh-CN' too */
-    strtolower(lang);
-
-    /* Search ":<lang>:" in supported languages string */
-    strncpy(match_str, ":", 1);
-    strncat(match_str, lang, LANG_MAXLEN);
-    strncat(match_str, ":\0", 2);
-
-    if (strstr(supported_languages, match_str))
-        is_supported = 1;
-
-    return is_supported;
-}
-
-/* Used by qsort() below */
-int sort_by_q(const void *x, const void *y) {
-    struct lang_list *a = (struct lang_list *)x;
-    struct lang_list *b = (struct lang_list *)y;
-    if (a->q > b->q) return -1;
-    if (a->q < b->q) return 1;
-    return 0;
-}
-
-/* Reads Accept-Language, parses it, and finds the first match
-   among the supported languages. In case of no match,
-   returns the default language.
-*/
-void select_language(const vcl_string *incoming_header, char *lang) {
-
-    struct lang_list pl[LANG_LIST_SIZE];
-    vcl_string *lang_tok = NULL;
-    vcl_string root_lang[3];
-    vcl_string *header;
-    vcl_string header_copy[HDR_MAXLEN];
-    vcl_string *pos = NULL;
-    vcl_string *q_spec = NULL;
-    unsigned int curr_lang = 0, i = 0;
-    float q;
-
-    /* Empty or default string, return default language immediately */
-    if (
-        !incoming_header
-        || (0 == strcmp(incoming_header, "en-US"))
-        || (0 == strcmp(incoming_header, "en-GB"))
-        || (0 == strcmp(incoming_header, DEFAULT_LANGUAGE))
-        || (0 == strcmp(incoming_header, ""))
-    )
-        RETURN_DEFAULT_LANG;
-
-    /* Tokenize Accept-Language */
-    header = strncpy(header_copy, incoming_header, sizeof(header_copy));
-
-    while ((lang_tok = strtok_r(header, " ,", &pos))) {
-
-        q = 1.0;
-
-        if ((q_spec = strstr(lang_tok, ";q="))) {
-            /* Truncate language name before ';' */
-            *q_spec = '\0';
-            /* Get q value */
-            sscanf(q_spec + 3, "%f", &q);
-        }
-
-        /* Wildcard language '*' should be last in list */
-        if ((*lang_tok) == '*') q = 0.0;
-
-        /* Push in the prioritized list */
-        PUSH_LANG(lang_tok, q);
-
-        /* For cases like 'en-GB', we also want the root language in the final list */
-        if ('-' == lang_tok[2]) {
-            root_lang[0] = lang_tok[0];
-            root_lang[1] = lang_tok[1];
-            root_lang[2] = '\0';
-            PUSH_LANG(root_lang, q - 0.001);
-        }
-
-        /* For strtok_r() to proceed from where it left off */
-        header = NULL;
-
-        /* Break out if stored max no. of languages */
-        if (curr_lang >= LANG_LIST_SIZE)
-            break;
-    }
-
-    /* Sort by priority */
-    qsort(pl, curr_lang, sizeof(struct lang_list), &sort_by_q);
-
-    /* Match with supported languages */
-    for (i = 0; i < curr_lang; i++) {
-        if (is_supported(pl[i].lang))
-            RETURN_LANG(pl[i].lang);
+  <%= if_app(app, backend) %>
+    if (false) {
```

Comment:
The template-generated conditional-blocks here was pretty complicated. I've refactored this stuff into an `if_else` helper method that separates the conditional-block string-generation from the underlying code blocks.